### PR TITLE
upgrades clojure version to 1.10

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- upgrades clojure version to 1.10

## Why are we making these changes?

Running the latest version of Clojure simplifies our move to JDK9/10 when required.

### CODE REVIEW

**Comment by shamsimam** [comment]
Un-wip-ed the PR :)

File: waiter/src/waiter/scheduler/shell.clj [hunk 1]
```diff
@@ -759,12 +759,13 @@
   "Detects and marks lost processes."
   [running-pids id->instance]
   (pc/map-vals
-    (fn [{:strs [killed? shell-scheduler/pid] :as instance}]
-      (cond-> (-> (pc/map-keys keyword instance)
-                  (update :started-at str-to-date-safe))
-        (and (not killed?) (not (contains? running-pids pid)))
-        (assoc :killed? true
-               :message "Process lost after restart")))
+    (fn [{:strs [killed?] :as instance}]
```
Comment (DaoWen) [inline waiter/src/waiter/scheduler/shell.clj]:
```suggestion
    (fn [{:strs [killed?] pid "shell-scheduler/pid" :as instance}]
```

File: waiter/src/waiter/scheduler/shell.clj [hunk 2]
```diff
@@ -759,12 +759,13 @@
...
+      (let [pid (get instance "shell-scheduler/pid")]
```
Comment (DaoWen) [inline waiter/src/waiter/scheduler/shell.clj]: if you move this back to the destructuring above, then we don't need this `let`.

Comment (shamsimam) [reply waiter/src/waiter/scheduler/shell.clj]: removed.

File: waiter/src/waiter/scheduler/shell.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/scheduler/shell.clj]: Done.

**Pull request merged**